src/main/kotlin/no/nav/klage/oppgave/service/distribusjon/KlagebehandlingDistribusjonService.kt

```diff
@@ -0,0 +1,76 @@
+package no.nav.klage.oppgave.service.distribusjon
+
+import no.nav.klage.oppgave.domain.klage.BrevMottaker
+import no.nav.klage.oppgave.domain.klage.Vedtak
+import no.nav.klage.oppgave.service.KlagebehandlingService
+import no.nav.klage.oppgave.util.getLogger
+import no.nav.klage.oppgave.util.getSecureLogger
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Propagation
+import org.springframework.transaction.annotation.Transactional
+import java.util.*
+
+@Service
+@Transactional
+class KlagebehandlingDistribusjonService(
+    private val klagebehandlingService: KlagebehandlingService,
+    private val vedtakDistribusjonService: VedtakDistribusjonService,
+    private val klagebehandlingAvslutningService: KlagebehandlingAvslutningService
+) {
+
+    companion object {
+        @Suppress("JAVA_CLASS_ON_COMPANION")
+        private val logger = getLogger(javaClass.enclosingClass)
+        private val secureLogger = getSecureLogger()
+    }
+
+    @Transactional(propagation = Propagation.REQUIRES_NEW)
+    fun distribuerKlagebehandling(klagebehandlingId: UUID) {
+        try {
+            val klagebehandling = klagebehandlingService.getKlagebehandlingForUpdate(klagebehandlingId, null)
+            logger.debug("1. Klagebehandlingversjon er ${klagebehandling.versjon}")
+            klagebehandling.vedtak
+                .filter { vedtak -> vedtak.erIkkeFerdigDistribuert() }
+                .forEach { vedtak ->
+                    logger.debug("Starter sjekk for om vedtak ${vedtak.id} skal distribueres")
+                    if (vedtak.harIngenBrevMottakere()) {
```

Comment:
Kan man tenke seg at mottakerlisten endrer seg mellom forsøk? Burde vi håndtert det? Vi har jo ikke støtte for det i GUI nå da.

Reply:
Per nå er det veldig unaturlig at det skal kunne skje. Bare en tanke jeg fikk. Tror koden slik den er vil holde.

Reply:
Jeg lurte veldig på hvor/når jeg skulle opprette BrevMottakere. Men kom vel frem til at iom at vi ikke har noe støtte for det i gui, så var det ikke så nøye. Men det må vi tenke på når den tid kommer.